library/src/scala/NamedTuple.scala

```diff
@@ -142,13 +139,14 @@ object NamedTupleDecomposition:
   extension [N <: Tuple, V <: Tuple](x: NamedTuple[N, V])
       /** The value (without the name) at index `n` of this tuple */
     inline def apply(n: Int): Tuple.Elem[V, n.type] =
-      inline x.toTuple match
-        case tup: NonEmptyTuple => tup(n).asInstanceOf[Tuple.Elem[V, n.type]]
-        case tup => tup.productElement(n).asInstanceOf[Tuple.Elem[V, n.type]]
+      x.toTuple.apply(n).asInstanceOf[Tuple.Elem[V, n.type]]
 
     /** The number of elements in this tuple */
     inline def size: Tuple.Size[V] = x.toTuple.size
 
+    /** The first element value of this tuple */
+    inline def head: Tuple.Elem[V, 0] = apply(0)
```

Comment:
so this one I'm sure doesn't need to move purely because it is a forwarder method

Reply:
unless we want to establish a rule that *all* methods go there